src/AuthenticatedAPIClient/tests/AuthenticatedAPIClient.test.jsx

```diff
@@ -1,428 +1,596 @@
+// encode mock JWT tokens
+Object.keys(jwtTokens).forEach((jwtTokenName) => {
+  const decodedJwt = jwtTokens[jwtTokenName].decoded;
+  jwtTokens[jwtTokenName].encoded = `header.${btoa(JSON.stringify(decodedJwt))}`;
+});
+
+const mockCsrfToken = 'thetokenvalue';
+const mockApiEndpointPath = `${process.env.BASE_URL}/api/v1/test`;
+
+window.location.assign = jest.fn();
+const mockCookies = new Cookies();
+
+// This sets the mock adapter on the default instance
+const axiosMock = new MockAdapter(axios);
+const accessTokenAxios = axios.create();
+const accessTokenAxiosMock = new MockAdapter(accessTokenAxios);
+AccessToken.__Rewire__('httpClient', accessTokenAxios); // eslint-disable-line no-underscore-dangle
+const csrfTokensAxios = axios.create();
+const csrfTokensAxiosMock = new MockAdapter(csrfTokensAxios);
+CsrfTokens.__Rewire__('httpClient', csrfTokensAxios); // eslint-disable-line no-underscore-dangle
+
+
 const client = getAuthenticatedAPIClient(authConfig);
-const originalGetDecodedAccessToken = client.getDecodedAccessToken;
-const originalIsAccessTokenExpired = client.isAccessTokenExpired;
 
-beforeEach(() => {
-  client.getDecodedAccessToken = originalGetDecodedAccessToken;
-  client.isAccessTokenExpired = originalIsAccessTokenExpired;
-  jest.spyOn(global.document, 'referrer', 'get').mockReturnValue('');
-});
+// Helpers
+const setJwtCookieTo = (jwtCookieValue) => {
+  mockCookies.get.mockReturnValue(jwtCookieValue);
+};
 
-function expectRefreshAccessTokenToHaveBeenCalled() {
-  expect(client.refreshAccessToken).toHaveBeenCalled();
-}
+const setJwtTokenRefreshResponseTo = (status, jwtCookieValue) => {
+  accessTokenAxiosMock.onPost().reply(() => {
+    setJwtCookieTo(jwtCookieValue);
+    return [status];
+  });
+};
 
-function expectRefreshAccessTokenToNotHaveBeenCalled() {
-  expect(client.refreshAccessToken).not.toHaveBeenCalled();
-}
 
-function expectGetCsrfTokenToHaveBeenCalled() {
-  expect(client.getCsrfToken).toHaveBeenCalled();
+function expectLogout(redirectUrl = process.env.BASE_URL) {
+  const encodedRedirectUrl = encodeURIComponent(redirectUrl);
+  expect(window.location.assign)
+    .toHaveBeenCalledWith(`${process.env.LOGOUT_URL}?redirect_url=${encodedRedirectUrl}`);
 }
 
-function expectGetCsrfTokenToNotHaveBeenCalled() {
-  expect(client.getCsrfToken).not.toHaveBeenCalled();
+function expectLogin(redirectUrl = process.env.BASE_URL) {
+  const encodedRedirectUrl = encodeURIComponent(redirectUrl);
+  expect(window.location.assign)
+    .toHaveBeenCalledWith(`${process.env.LOGIN_URL}?next=${encodedRedirectUrl}`);
 }
 
-function expectCsrfHeaderSet(request) {
-  expect(request.headers['X-CSRFToken']).toBe('test-csrf-token');
-}
+const expectLogErrorToHaveBeenCalledWithMessage = (callParams, errorMessage, customAttributes) => {
+  const loggedError = callParams[0];
+  expect(loggedError.message).toEqual(errorMessage);
+  if (customAttributes) {
+    expect(callParams[1]).toEqual(customAttributes);
```

Comment:
An you move the objectContaining expect call into this helper? It would better match the argument name of customAttributes, and clean up the tests.

Reply:
I could, but we do have more clarity around what exactly we want in there by leaving it in the tests. { cookieValue } for example is supplied in one log instead of the axios request stuff.

Reply:
Done